### ISSUE TITLE
models/export.py setting model.train() may be changed to false in onnx

### ISSUE THREAD

**Issue by ChaofWang** [ultralytics/yolov5#3346]
## 🐛 Bug

I think you may want to have no grid construction in [Detect](https://github.com/ultralytics/yolov5/blob/aad99b63d6ac63278021a66f8a096e5232ed24b0/models/yolo.py#L50) layer  by setting [model.train()](https://github.com/ultralytics/yolov5/blob/aad99b63d6ac63278021a66f8a096e5232ed24b0/models/export.py#L62) when export model to onnx with --train. But in "torch.onnx.export", model.training seems to be reset to mode.training=False by default.  

In fact, the param 'training' in the "torch.onnx.export" can be set to "training=torch.onnx.TrainingMode.TRAINING", so that the model can be set to train for conversion instead of using model.train().  But it doesn't seem to be a good way to be recommended. And I found that this part was correct before version 4.0, but after version 5.0, the "export" was deleted in Detect. 

## To Reproduce (REQUIRED)

Input:
```
python models/export.py --weights yolov5s.pt --img 640 --batch 1 --train
```

Output:
```
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
```

## Expected behavior
This part should be skipped. https://github.com/ultralytics/yolov5/blob/aad99b63d6ac63278021a66f8a096e5232ed24b0/models/yolo.py#L50-L62

## Environment

 - OS: Ubuntu18.04
 - GPU: 3080
 - PyTorch 1.8.0+cu111
 -  onnx 1.9.0

**Comment by glenn-jocher** [comment]
@ChaofWang hi, thanks for the bug report! You are correct, ONNX export is not operating in train mode for some reason. It seems the ONNX exporter is forcing it back into eval mode.

**Comment by glenn-jocher** [comment]
@ChaofWang I've implemented a solution per your recommendation by adding two arguments to the onnx export function in export.py L100:
```python
training=torch.onnx.TrainingMode.TRAINING if opt.train else torch.onnx.TrainingMode.EVAL,
do_constant_folding=not opt.train,
```

Results are here:
<img width="906" alt="Screenshot 2021-05-26 at 12 27 07" src="https://user-images.githubusercontent.com/26833433/119644900-ba11f480-be1d-11eb-9e3c-9e01e610195f.png">

If this solution works for you **please submit a PR with this update**, thank you!

**Comment by glenn-jocher** [comment]
TODO: ONNX export in .train() mode fix

**Comment by ChaofWang** [comment]
@glenn-jocher hi, this solution works for me. I have submitted PR for this update

**Comment by glenn-jocher** [comment]
@ChaofWang good news 😃! Your original issue may now be fixed ✅ in PR #3362. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
